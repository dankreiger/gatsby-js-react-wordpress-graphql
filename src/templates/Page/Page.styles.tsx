import styled from 'styled-components'

export const PageTitle = styled.h1`
  margin-bottom: 50px;
`
export const PageContent = styled.div`
  p {
    margin-bottom: 30px;
    line-height: 1.4;
    font-size: 1.4em;
  }
`