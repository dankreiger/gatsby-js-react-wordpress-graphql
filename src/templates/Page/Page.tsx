import React from 'react'
import he from 'he'
import { PageNode } from '../../typings/QueryProps.interface'
import Layout from '../../components/Layout/Layout'
import { PageTitle, PageContent } from './Page.styles'
import SiteSlider from '../../components/SiteSlider/SiteSlider'

export default ({ pageContext }: { pageContext: PageNode }) => {
  let videoIds: string[] = he
    .decode(pageContext.content)
    .replace(/(<([^>]+)>)/gi, '')
    .split(',')
    .map((videoId: string) => videoId.trim())
  return (
    <Layout>
      <PageTitle dangerouslySetInnerHTML={{ __html: pageContext.title }} />
      {pageContext.slug !== 'videos' && (
        <PageContent
          dangerouslySetInnerHTML={{ __html: pageContext.content }}
        />
      )}
      {pageContext.slug === 'videos' && <SiteSlider videoIds={videoIds} />}
    </Layout>
  )
}
