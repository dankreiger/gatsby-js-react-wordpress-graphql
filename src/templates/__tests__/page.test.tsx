import * as React from "react"
import renderer from "react-test-renderer"

import Page from "../Page/Page"

describe("Page", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Page
          pageContext={{ id: "1234", content: "content", title: "some title" }}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
