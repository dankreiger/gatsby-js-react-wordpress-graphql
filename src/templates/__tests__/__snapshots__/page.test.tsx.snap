// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div>
  <h1>
    some title
  </h1>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "content",
      }
    }
  />
</div>
`;
